### PR TITLE
Add `Structure` property

### LINKED ISSUE #468: Add `Structure` property
### Problem Description
_As a user, I'd like a property that checks to see whether the overall structure of the synthetic data matches the real data_

Once we create the TableFormat metric (#463), we'd like to use it to calculate this new property.

### Expected behavior
- Add a new property `Structure` for single and multi table reports.
- The multi-table property can just call the single table property for each table.

#### Attributes
The property should have the following attributes:
- `_num_iteration_case = 'table'`

#### Score Calculation
This property should use the `TableFormat` metric to calculate the score for each table. The metric should ignore checking the dtype for ID and PII columns by passing them to the metric using the `ignore_dtype_columns` parameter. 

#### Methods
The property should define the following methods:
- `_generate_details(self, real_data, synthetic_data, metadata, progress_bar=None)`: This method will generate the `details` dataframe for the property. 
    - The details dataframe should have the following columns:
        - `Metric`: The metric that is applied to the given column
        - `Score`: The score returned from the metric
        - `Error`: The error that occurred while calculating the metric, if it exists. 
    - The progress bar should update for the table has been evaluated.
- `get_visualization(self)`: This method should return a bar graph of the scores for each column, similar to other properties.
    - Label the bars using the column name
    - On hover, there should be more details about the metric applied and the score

### PR DESCRIPTION
**Pull request by R-Palazzo**
CU-86ayg9qwy
Resolve #468

Let me know if this visualization works @frances-h, @npatki
Screenshot 2023-10-26 at 08 23 20

### CODE REVIEW

**Comment by npatki** [comment]
@R-Palazzo seems OK for an MVP.

File: sdmetrics/reports/single_table/_properties/structure.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+        Returns:
+            plotly.graph_objects._figure.Figure
+        """
+        average_score = self._compute_average()
+
+        fig = px.bar(
+            data_frame=self.details.dropna(subset=['Score']),
+            y='Score',
+            title=f'Data Diagnostics: Structure (Average Score={round(average_score, 2)})',
```
Comment (npatki) [inline sdmetrics/reports/single_table/_properties/structure.py:82]: @R-Palazzo Title should be **Data Diagnostic** (singular, not plural)

We should make sure the title of these visualizations are consistent everywhere

**Review by npatki — CHANGES_REQUESTED** [review]
@R-Palazzo one small title change. I also wonder if you can share a screenshot of the tooltip (on-hover) to verify that it makes sense for the visualization.

File: sdmetrics/reports/single_table/_properties/structure.py (hunk 1 above)
Comment (R-Palazzo) [reply sdmetrics/reports/single_table/_properties/structure.py]: Good catch, fixed in [620285f](https://github.com/sdv-dev/SDMetrics/pull/483/commits/620285f1f92ec9e57d6e5128d32d2cbabf16afbe)

**Review by frances-h — APPROVED** [review]

**Review by fealho — APPROVED** [review]
Looks pretty good 👍

**Pull request merged**